components/core/Component/Overlay/OverlayTrigger.razor.cs

```diff
@@ -338,7 +338,7 @@ protected virtual async Task OnTriggerContextmenu(MouseEventArgs args)
 
         protected virtual void OnMouseUp(JsonElement element)
         {
-            if (_mouseInOverlay == false && _mouseInTrigger == false)
+            if (_mouseInOverlay == false && _mouseInTrigger == false && Visible)
```

Comment:
If you use `Visible` as a condition, `DatePicker` component may be affected so that the date selector can not be closed.